File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        curMax = ans = nums[0]
        for num in nums[1:]:
            curMax = max(num+curMax,num)
            ans = max(ans,curMax)
        return ans
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSofar = 0;
        int maxi=INT_MIN;
        for(int i=0;i<nums.size();i++){
            maxSofar = max(maxSofar+nums[i], nums[i]);
            maxi=max(maxi,maxSofar);
        }
        return maxi;
    }
};